src/glean/private/event.h

```diff
@@ -0,0 +1,60 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef EVENT_METRIC_H
+#define EVENT_METRIC_H
+
+#include "vpnglean.h"
+#if defined(UNIT_TEST)
+#  include "glean/glean.h"
+#  include <QJsonArray>
+#  include <QJsonDocument>
+#endif
+
+template <class T>
+class EventMetric final {
+ public:
+  constexpr explicit EventMetric(int aId) : m_id(aId) {}
+
+  // Coudn't figure out how to have these declaredi n a cpp file and still have
```

Comment:
I don't understand this comment. What do you want to achieve?

Reply:
I wanted to have the function implementation on the a .cpp files. Templates are not accessible in the .cpp files though.